pages/dao/[symbol]/proposal/components/instructions/raydium/RemoveLiquidity.tsx

```diff
@@ -0,0 +1,232 @@
+import { GovernedMultiTypeAccount } from '@utils/tokens'
+import Select from '@components/inputs/Select'
+import {
+  ProgramAccount,
+  serializeInstructionToBase64,
+  Governance,
+} from '@solana/spl-governance'
+import GovernedAccountSelect from '../../GovernedAccountSelect'
+import { getLPMintInfo } from '@tools/sdk/raydium/helpers'
+import { liquidityPoolKeysList } from '@tools/sdk/raydium/poolKeys'
+import { createRemoveLiquidityInstruction } from '@tools/sdk/raydium/createRemoveLiquidityInstruction'
+import BigNumber from 'bignumber.js'
+import { jsonInfo2PoolKeys } from '@raydium-io/raydium-sdk'
+
+const RemoveLiquidityRaydium = ({
+  index,
+  governance,
+}: {
+  index: number
+  governance: ProgramAccount<Governance> | null
+}) => {
+  const connection = useWalletStore((s) => s.connection)
+  const wallet = useWalletStore((s) => s.current)
+  const { realmInfo } = useRealm()
+  // const { getGovernancesByAccountType } = useGovernanceAssets()
+  const [governedAccounts, setGovernedAccounts] = useState<
+    GovernedMultiTypeAccount[]
+  >([])
+  const {
+    governancesArray,
+    governedTokenAccounts,
+    getMintWithGovernances,
+  } = useGovernanceAssets()
+
+  useEffect(() => {
+    async function prepGovernances() {
+      const mintWithGovernances = await getMintWithGovernances()
+      const matchedGovernances = governancesArray.map((gov) => {
+        const governedTokenAccount = governedTokenAccounts.find(
+          (x) => x.governance?.pubkey.toBase58() === gov.pubkey.toBase58()
+        )
+        const mintGovernance = mintWithGovernances.find(
+          (x) => x.governance?.pubkey.toBase58() === gov.pubkey.toBase58()
+        )
+        if (governedTokenAccount) {
+          return governedTokenAccount as GovernedMultiTypeAccount
+        }
+        if (mintGovernance) {
+          return mintGovernance as GovernedMultiTypeAccount
+        }
+        return {
+          governance: gov,
+        }
+      })
+      setGovernedAccounts(matchedGovernances)
+    }
+    prepGovernances()
+  }, [])
+
+  const shouldBeGoverned = index !== 0 && governance
+  const programId: PublicKey | undefined = realmInfo?.programId
+  const [form, setForm] = useState<RemoveLiquidityRaydiumForm>({
+    governedAccount: undefined,
+    liquidityPool: '',
+    amountIn: 0,
+  })
+  const [formErrors, setFormErrors] = useState({})
+  const { handleSetInstructions } = useContext(NewProposalContext)
+
+  const [maxLPAmount, setMaxLPAmount] = useState(0)
+  const [LPDecimals, setLPDecimals] = useState(9)
+  useEffect(() => {
+    const fetchLpData = async () => {
+      if (!form.governedAccount?.governance.pubkey || !form.liquidityPool)
+        return
+      const poolKeys = liquidityPoolKeysList[form.liquidityPool]
+
+      const { maxBalance, decimals } = await getLPMintInfo(
+        connection,
+        new PublicKey(poolKeys.lpMint),
+        form.governedAccount.governance.pubkey
+      )
+      setMaxLPAmount(maxBalance)
```

Comment:
Looks like maxBalance and decimals are important/required in here.

IMO I would create a unique state and make handle explicitly _null_ data.

Example:

```
  const [lpMintInfo, setLpMintInfo] = useState<{
    maxBalance: number
    decimals: number
  } | null>(null)
```

And for example in the display:

```
<Input
        label={`LP Token Amount to withdraw - max: ${lpMintInfo ? lpMintInfo.maxLPAmount : "-"}`}
```

But that's only my opinion, it works as it is now 👍

Reply:
yes, since we don't want to have one without the other